spec/unit/http/session_spec.rb

```diff
@@ -83,4 +83,58 @@ def check_connection?(session, service, ssl_context: nil)
       }.to raise_error(ArgumentError, "Unknown service westbound")
     end
   end
+
+  context 'when resolving using multiple resolvers' do
+    let(:session) { client.create_session }
+
+    it "prefers SRV records" do
```

Comment:
@josh I removed the test `it "resolves once per session"` since I feel like we're already testing that behavior below

Reply:
ah I was thinking of the case where two calls to `session.route_to(:ca)` should only cause one simple service status request.

Reply:
Ah right, https://github.com/puppetlabs/puppet/blob/b616178de85473c77e0821ec029faf6d1dc07fe7/spec/unit/http/session_spec.rb#L52, got it! 👍